runtime/vm/MHInterpreter.cpp

```diff
@@ -1353,6 +1366,127 @@ VM_MHInterpreter::getArgSlotsBeforePosition(j9object_t argumentTypes, U_32 argPo
+	j9object_t argumentTypes = getMethodTypeArguments(filterHandleType);
+	U_32 filterArgSlots = getMethodTypeArgSlots(filterHandleType);
+	j9object_t argumentIndices = J9VMJAVALANGINVOKEFILTERARGUMENTSWITHCOMBINERHANDLE_ARGUMENTINDICES(_currentThread, methodHandle);
+	U_32 argumentIndicesCount = J9INDEXABLEOBJECT_SIZE(_currentThread, argumentIndices);
+	UDATA *spFirstFilterArgSlot = _currentThread->sp + filterArgSlots;
+
+	/* [... filterHandle args descriptionBytes MethodTypeFrame] */
+	UDATA *spPriorToFrameBuild = _currentThread->sp;
+	(void)buildMethodTypeFrame(_currentThread, filterHandleType);
+
+	/* [... filterHandle args descriptionBytes MethodTypeFrame filterHandle args] */
+	_currentThread->sp -= (filterArgSlots  + 1);
+	memcpy(_currentThread->sp, spPriorToFrameBuild, sizeof(UDATA) * (filterArgSlots+1));
+
+	/* [... filterHandle args descriptionBytes MethodTypeFrame filterHandle PlaceHolderFrame filterHandle args] */
+	insertPlaceHolderFrame(filterArgSlots, methodHandle, J9VMJAVALANGINVOKEMETHODHANDLE_FILTERARGUMENTSWITHCOMBINERPLACEHOLDER_METHOD(_vm));
+
+	/* Get combinerHandle */
+	j9object_t combinerHandle = getCombinerHandleForFilter(methodHandle);
+
+	/* Replace filterHandle with combinerHandle and adjust sp according to combinerHandle.type.argSlots */
+	/* [... filterHandle args descriptionBytes MethodTypeFrame filterHandle PlaceHolderFrame combinerHandle filterArgs] */
+	((j9object_t *)_currentThread->sp)[filterArgSlots] = combinerHandle;
+	UDATA *spCombinerSlot = _currentThread->sp + filterArgSlots;
+
+	/* Copy all arguments specified by argumentIndices from filterHandle to the stack slots used by combinerHandler */
+	U_32 arrayIndex = 0;
+	for (arrayIndex = 0; arrayIndex < argumentIndicesCount; arrayIndex++) {
+		U_32 argumentTypeIndex = (U_32)J9JAVAARRAYOFINT_LOAD(_currentThread, argumentIndices, arrayIndex);
+
+		/* Determine the slot index of each argument specified by the index in argumentIndices */
+		U_32 argumentTypeSlots = getArgSlotsBeforePosition(argumentTypes, argumentTypeIndex);
+
+		/* Determine the type of each argument specified by argumentIndices before copying.
+		* Note: long/double type takes 2 slots while other types take 1 slot.
+		*/
+		j9object_t argumentTypeAtIndex =
+			_objectAccessBarrier->inlineIndexableObjectReadObject(
+				_currentThread,
+				argumentTypes,
+				argumentTypeIndex);
+		J9Class *argumentClassAtIndex = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, argumentTypeAtIndex);
+
+		if ((_vm->doubleReflectClass == argumentClassAtIndex)
+			|| (_vm->longReflectClass == argumentClassAtIndex)
+		) {
+			spCombinerSlot -= 2;
+			*(U_64*)spCombinerSlot = *(U_64*)(spFirstFilterArgSlot - argumentTypeSlots - 2);
+		} else {
+			spCombinerSlot -= 1;
+			*spCombinerSlot = *(spFirstFilterArgSlot - argumentTypeSlots - 1);
+		}
+	}
+	_currentThread->sp = spCombinerSlot; /* after loop spCombinerSlot is at the top of the combiner arguments */
```

Comment:
This shouldn't be necessary as `UDATA *spCombinerSlot = _currentThread->sp + filterArgSlots;`

Maybe add an assert that `_currentThread->sp == spCombinerSlot`?

Reply:
It's necessary to do this here because the sp is previously set at the number of arguments for the "next" handle after the memcpy. This sets it to the correct number of arguments for the combiner handle which might be different. I will apply this comment to the version I'm working on without the memcpy though.